addons/mail/static/src/new/core/link_preview_model.js

```diff
@@ -1,34 +1,33 @@
 /** @odoo-module */
 
-/**
- * @class LinkPreview
- */
 export class LinkPreview {
+    /** @type {Number} */
+    id;
+    /** @type {Object} */
+    message;
+    /** @type {string} */
+    image_mimetype;
+    /** @type {string} */
+    og_description;
+    /** @type {string} */
+    og_image;
+    /** @type {string} */
+    og_mimetype;
+    /** @type {string} */
+    og_title;
+    /** @type {string} */
+    og_type;
+    /** @type {string} */
+    og_source_url;
+
     /**
      * @param {Object} data
+     * @returns {LinkPreview}
      */
-    constructor(data) {
-        /** @type {Number} */
-        void this.id;
-        /** @type {Object} */
-        void this.message;
-        /** @type {string} */
-        void this.image_mimetype;
-        /** @type {string} */
-        void this.og_description;
-        /** @type {string} */
-        void this.og_image;
-        /** @type {string} */
-        void this.og_mimetype;
-        /** @type {string} */
-        void this.og_title;
-        /** @type {string} */
-        void this.og_type;
-        /** @type {string} */
-        void this.og_source_url;
-
-        // Assign every field from the server RPC to the class
-        Object.assign(this, data);
+    static insert(data) {
+        const linkPreview = new LinkPreview();
+        Object.assign(linkPreview, data);
+        return linkPreview;
```

Comment:
If we're not actually inserting the preview into the global state I would not make an insert method for now and just keep a constructor.